test/FrequencyMapAgentTest.cpp

```diff
@@ -485,3 +486,108 @@ TEST_F(FrequencyMapAgentTest, validate_policy)
+        key_value.second.erase(std::remove(key_value.second.begin(), key_value.second.end(), '"'), key_value.second.end());
+        EXPECT_EQ(key_value.second, reference_map);
+    }
+}
+
+TEST_F(FrequencyMapAgentTest, test_default_freq_hash)
+{
+    FrequencyMapAgent frequency_agent(
+        {},
+        {
+            {0x00000000a74bbf35,
+             0x00000000d691da00,
+             0x8000000000000000}
+        },
+        *m_platform_io, *m_platform_topo
+    );
+    std::string reference_map =  "{0x00000000a74bbf35: null, "
+                                  "0x00000000d691da00: null, "
+                                  "0x8000000000000000: null}";
+
+    auto result = frequency_agent.report_host();
+    for (auto& key_value : result) {
+        key_value.second.erase(std::remove(key_value.second.begin(), key_value.second.end(), '"'), key_value.second.end());
+        EXPECT_EQ(key_value.second, reference_map);
+    }
+}
+
+TEST_F(FrequencyMapAgentTest, test_both_map_and_set)
+{
+    FrequencyMapAgent frequency_agent(
+        {
+            {0x000000003ddc81bf, 1000000000},
+            {0x00000000644f9787, 2100000000},
+            {0x00000000725e8066, 2100000000},
+            {0x000000007b561f45, 2100000000},
+            {0x00000000a74bbf35, 1200000000},
+            {0x00000000d691da00, 1900000000},
+            {0x8000000000000000, 2100000000}
+        },
+        {
+            {0x00000000644f9789,
+             0x000000007b561f47,
+             0x00000000d691da02}
+        },
+        *m_platform_io, *m_platform_topo
+    );
+    std::string reference_map =  "{0x000000003ddc81bf: 1000000000, "
+                                  "0x00000000644f9787: 2100000000, "
+                                  "0x00000000644f9789: null, "
+                                  "0x00000000725e8066: 2100000000, "
+                                  "0x000000007b561f45: 2100000000, "
+                                  "0x000000007b561f47: null, "
+                                  "0x00000000a74bbf35: 1200000000, "
+                                  "0x00000000d691da00: 1900000000, "
+                                  "0x00000000d691da02: null, "
+                                  "0x8000000000000000: 2100000000}";
+
+    auto result = frequency_agent.report_host();
+    for (auto& key_value : result) {
+        key_value.second.erase(std::remove(key_value.second.begin(), key_value.second.end(), '"'), key_value.second.end());
+        EXPECT_EQ(key_value.second, reference_map);
+    }
+}
+
+TEST_F(FrequencyMapAgentTest, test_neither_map_nor_set)
+{
+    FrequencyMapAgent frequency_agent(
+        {},
+        {},
+        *m_platform_io, *m_platform_topo
+    );
+    std::string reference_map =  "{}";
+
+    auto result = frequency_agent.report_host();
+    for (auto& key_value : result) {
+        key_value.second.erase(std::remove(key_value.second.begin(), key_value.second.end(), '"'), key_value.second.end());
```

Comment:
Same question here as before: what purpose does this line serve?  The map is empty.

Reply:
This line removes the extra `"` characters from the string. Previously the keys used to be strings surrounded with quotes: `"0x01234567"` Now they are just `0x01234567`